### PR TITLE
[meecast.spec] Set all version numbers coherently to 1.10.40

### PR DESCRIPTION
**Pull request by Olf0**
Because the SailfishOS-OBS, which compiles all packages for the SailfishOS:Chum repository, sets all version numbers of sub-packages to the one of the main package, this version number is a merger of the one of the `MeeCast daemon` (`1.10.1`) and the one of `MeeCast` proper (`1.1.40`) in order to avoid confusion among users and package managers (RPM, Zypper, pkcon) when both, [Vasvlad's repository at OpenRepos](https://openrepos.net/user/678/programs) and the [SailfishOS:Chum repository are](https://build.sailfishos.org/project/show/sailfishos:chum) subscribed.

### CODE REVIEW

**Review by vasvlad — APPROVED** [review]
maybe it should be

**Comment by Olf0** [comment]
I just checked again, see
meecast_sfos-chum_screenshot_2024-09-09

Hence currently the MeeCast daemon binary has a lot lower version number at SailfishOS:Chum (`1.1.40.2`) than its binary at OpenRepos (`1.10`) which is compiled from the same source code snapshot.  For MeeCast event and lockscreen it is the other way around: `1.1` / `0.5` at OpenRepos and `1.1.40.2` for both at SailfishOS:Chum, despite being built from the same source code.

So a common version number for all (sub-)packages has to be higher than the highest current one: `1.10`
In order to signal some continuity to MeeCast `1.1.40`, I still believe using `1.10.40` for the next release makes (some / a little) sense (or rather: anything else would be even more confusing).

-------------------------------------------------

### Git tag processing and using it to construct a `Version:` number by `tag_git`

When I asked, I was told a lot of hearsay and cooking-recipes, how `tag_git` processes a Git tag to construct a `Version:` number from it, which is then used to overwrite the original `Version:` number in the spec file before building the package.  As I do not like to be told "do this, this works for me" without understanding why, I took a closer look, analysed Jolla's `tag_git` OBS module (it is a shell script) and wrote [this README.md](https://github.com/MeeGoIntegration/obs-service-tar-git/blob/62f79306b06a2610c3867c0bccce14f2d8019676/README.md) (unfortunately, [its pull request](https://github.com/MeeGoIntegration/obs-service-tar-git/pull/4) was actively stalled).

From this knowledge I deducted these [requirements for Git tags](https://github.com/storeman-developers/harbour-storeman-installer/wiki/Git-tag-format#requirements).

The only Git tag format which fulfils these requirements *and* can be used to construct a valid path to the automatically generated source archive at GitHub (when setting a Git tag there) is `%{release}/%{version}`!
Mind that fulfilling these requirements is a **must** for the resulting version numbers for builds from a CI/CD pipeline at GitHub and from the SailfishOS-OBS (e.g. in the SailfishOS:Chum repository) to be the same.

### TL;DR
Please do solely use `%{release}/%{version}` as Git tag format<sup>**\***</sup>; at least for release versions (i.e. versions which are also built at the SailfishOS-OBS), but I personally prefer to consistently use this format for all Git tags to avoid confusion (one's own and of others).
<sub>The release titles at GitHub are free text, but I suggest using their simplest, usual form: `%{version}-%{release}`, i.e. without a prefixed, superfluous `v` which transports no additional information).</sub>

**\***: After a pull request is merged which will adapt the `build.yaml` workflow configuration file accordingly and will also introduce the changes from this branch.

**Comment by Olf0** [comment]
This branch was merged by https://github.com/Meecast/meecast/pull/65 (i.e. to upstream branch `sailfishos`).

Unfortunately switching the base branch multiple times seems to have confused git(hub).  Nevermind!

**Comment by Olf0** [comment]
This PR is stale, broken by the many experimental base changes, and superseded by https://github.com/Meecast/meecast/pull/65.
